src/btree/row_modify.c

```diff
@@ -429,7 +429,7 @@ __wt_update_obsolete_check(
             first = NULL;
 
         /* Cannot truncate the updates if we need to remove the updates from the history store. */
-        if (F_ISSET(upd, WT_UPDATE_TO_DELETE_FROM_HS))
+        if (F_ISSET(upd, WT_UPDATE_HS_MAX_STOP))
```

Comment:
We do use this flag to remove the update from HS even if it has the stop also because the data store is rolled back. Using the flag HS_MAX_STOP may be confusing?

Reply:
We no longer mark them to be deleted. Reconciliation itself will decide which to delete which not so I introduced this flag to distinguish with a proper hs record with a valid stop time point

Reply:
ok.